### PR TITLE
build(lint): switch over to ruff; add taplo

### PR DESCRIPTION
**Pull request by yajo**
Ruff is faster, better integrated, easier to configure... Cool!

Taplo also helps with toml maintenance.

### CODE REVIEW

**Comment by sisp** [comment]
There's some Nix error, but I still don't know enough about Nix to say what's wrong. 🫣

And I wonder whether the Ruff pre-commit hook also checks/applies formatting because the [hook spec](https://github.com/cachix/pre-commit-hooks.nix/blob/e5ee5c5f3844550c01d2131096c7271cec5e9b78/modules/hooks.nix#L1040-L1046) looks a bit outdated. With a recent Ruff version, it's `ruff check` for linting and `ruff format` for formatting. We'd want to make sure the formatting is also checked through CI. Do you know, @yajo?

File: pyproject.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ copier = "copier.__main__:copier_app_run"
 "Bug Tracker" = "https://github.com/copier-org/copier/issues"
 
 [tool.poetry.dependencies]
-python = ">=3.8,<4.0" # HACK https://github.com/PyCQA/isort/issues/1945
+python = ">=3.8,<4.0"                                               # HACK https://github.com/PyCQA/isort/issues/1945
```
Comment (pawamoy) [inline pyproject.toml:30]: Wow, what's this extra space :joy:? Does it try to align comment on the far right or something?

Comment (yajo) [reply pyproject.toml:30]: Wow amazing, I didn't see this. Let me check

Comment (yajo) [reply pyproject.toml]: Fun, it turns out that the issue itself was closed long ago, so I removed the hack.

**Comment by yajo** [comment]
Well, nix configs make the `.pre-commit-config.yaml` nix-generated file use taplo. It's just a TOML formatter.

You have a brief explanation of how it works [here](https://devenv.sh/pre-commit-hooks/#managing-the-pre-commit-configyaml-file).

**Pull request merged**